server/friends.ts

```diff
@@ -67,13 +67,16 @@ function canPM(sender: User, receiver: User | null) {
 
 export class FriendsDatabase {
 	file: string;
-	cache: Cache<Set<string>>;
 	constructor(file: string = DEFAULT_FILE) {
 		this.file = file === ':memory:' ? file : path.resolve(file);
-		this.cache = new Cache<Set<string>>(async user => {
-			const data = await this.getFriends(user as ID);
-			return new Set(data.map(f => f.friend));
-		});
+	}
+	async updateCache(user: User) {
```

Comment:
I feel like `updateUser` or `updateUserCache` or `updateUserObject` might be a better name here; when I hear `updateCache` I would think of a central cache rather than the `User#friends` set.

Reply:
I renamed it to `updateUserCache`.